File: src/test/kotlin/chapter1/java/PresenterTests.java
package chapter1.java;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PresenterTests {

    private Presenter nat = new Presenter("1", "Nat");
    private Presenter duncan = new Presenter("2", "Duncan");

    @Test
    void properties() {
        assertEquals("1", nat.getId());
        assertEquals("Duncan", duncan.getName());
    }

    @Test
    void equality() {
        assertEquals(nat, new Presenter("1", "Nat"));
        assertNotEquals(nat, duncan);
    }

    @Test
    void testHashCode() {
        assertEquals(nat.hashCode(), new Presenter("1", "Nat").hashCode());
        assertNotEquals(nat.hashCode(), duncan.hashCode());
    }
}

File: src/test/kotlin/chapter1/kotlin/PresenterTests.kt
package chapter1.kotlin

import chapter1.java.Presenter
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class PresenterTests {

    private val nat = Presenter("1", "Nat")
    private val duncan = Presenter("2", "Duncan")

    @Test
    fun properties() {
        Assertions.assertEquals("1", nat.id)
        Assertions.assertEquals("Duncan", duncan.name)
    }

    @Test
    fun equality() {
        Assertions.assertEquals(nat, Presenter("1", "Nat"))
        Assertions.assertNotEquals(nat, duncan)
    }

    @Test
    fun testHashCode() {
        Assertions.assertEquals(nat.hashCode(), Presenter("1", "Nat").hashCode())
        Assertions.assertNotEquals(nat.hashCode(), duncan.hashCode())
    }
}